src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -801,9 +806,9 @@ func (c controller) waitingForClusterVersion() error {
 		return false
 	}
 
-	err := utils.WaitForPredicate(WaitTimeout, GeneralProgressUpdateInt, isClusterVersionAvailable)
+	err := utils.WaitForPredicateWithTimer(ctx, WaitTimeout, GeneralProgressUpdateInt, isClusterVersionAvailable)
```

Comment:
@tsorya is changing this very utility. You have a clash here. @tsorya this is the benefit of the DSL pattern that it does not care about adding dependencies